packages/mui-joy/src/Link/Link.tsx

```diff
@@ -0,0 +1,233 @@
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { OverridableComponent } from '@mui/types';
+import {
+  unstable_capitalize as capitalize,
+  unstable_useForkRef as useForkRef,
+  unstable_useIsFocusVisible as useIsFocusVisible,
+} from '@mui/utils';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import styled from '../styles/styled';
+import useThemeProps from '../styles/useThemeProps';
+import Typography from '../Typography';
+import linkClasses, { getLinkUtilityClass } from './linkClasses';
+import { LinkProps, LinkTypeMap } from './LinkProps';
+
+const useUtilityClasses = (ownerState: LinkProps) => {
+  const { component, level, color, variant, underline, focusVisible, disabled } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      color && `color${capitalize(color)}`,
+      disabled && 'disabled',
+      focusVisible && 'focusVisible',
+      level,
+      underline && `underline${capitalize(underline)}`,
+      variant && `variant${capitalize(variant)}`,
+      component === 'button' && 'button',
+    ],
+  };
+
+  return composeClasses(slots, getLinkUtilityClass, {});
+};
+
+const LinkRoot = styled(Typography, {
+  name: 'MuiLink',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: LinkProps }>(({ theme, ownerState }) => {
+  return [
+    {
+      ...(ownerState.underline === 'none' && {
+        textDecoration: 'none',
+      }),
+      ...(ownerState.underline === 'hover' && {
+        textDecoration: 'none',
+        '&:hover': {
+          textDecoration: 'underline',
+        },
+      }),
+      ...(ownerState.underline === 'always' && {
+        textDecoration: 'underline',
+        // textDecorationColor: color !== 'inherit' ? alpha(color, 0.4) : undefined,
+        '&:hover': {
+          textDecorationColor: 'inherit',
+        },
+      }),
+      // Same reset as ButtonBase.root
+      ...(ownerState.component === 'button' && {
+        position: 'relative',
+        WebkitTapHighlightColor: 'transparent',
+        backgroundColor: 'transparent', // Reset default value
+        // We disable the focus ring for mouse, touch and keyboard users.
+        outline: 0,
+        border: 0,
+        margin: 0, // Remove the margin in Safari
+        borderRadius: 0,
+        padding: 0, // Remove the padding in Firefox
+        cursor: 'pointer',
```

Comment:
I have this idea to make the `variant` undefined by default so that when you render `<Link />`, you get the simplest Link without hover & active state. Once `variant` is set, this color will be overriden.

```suggestion
        cursor: 'pointer',
        color: theme.vars.palette[ownerState.color!]?.textColor,
```